### DISCUSSION TITLE
Driver 'result' object and error handling?

### DISCUSSION THREAD

**Discussion by jhunterkohler** [Automattic/mongoose#10554] [General]
As you probably all know methods that more directly access the underlying driver like `Model.prototype.updateOne()`  result results similar to:

```json
{
    "ok": 1,
    "n": 1,
    "nModified": 1
}
```

I have two quick questions since I've been very confused on how to handle the raw results:

#### Does mongoose throw and error for `ok: 0`?

Obviously the Node.JS MongoDB driver results `ok: 0` if an error occurs while writing, but does Mongoose recognize this internally, or just pass back the result?

#### How should I handle this?

What is the typical pattern for handling this type of result? I feel that, due to repetitiveness of checking the response for such a simple operation, I am missing some patterned solution. Also, it seems to me that `ok: 0` would signify something that is out of my control to handle, because it is something wrong in the database itself, leading to confusion of wether I should notify users that a server error has occurred over, or if there could be some better way to get information.

**Comment by vkarpov15** [comment]
Does the MongoDB Node driver send back a result with `ok: 0`? I have never seen such a case, the MongoDB driver always throws an error if there's some error writing as far as I'm aware.

**Reply by jhunterkohler** [reply]
Ok, thank you. From my observations I believe thats 100% correct. Thanks for making a thorough project!
